concepts/basics/introduction.md

```diff
@@ -0,0 +1,111 @@
+# Introduction
+
+To set off the C++ journey we are starting with variables, function calls, and comments.
+
+## Comments
+
+Comments come in two flavors: single- and multi-line.
+Everything that comes after `//` on the same line is ignored by the compiler.
+Multi-line comments are also known as C-style comments. 
+They are surrounded by `/*` and `*/`.
+Anything that comes between these will be ignored as well.
+
+## Variables
+
+C++ is a typed language.
+All types need to be known at compile time, and so you generally need to state them explicitly.
+A variable's type cannot change.
+An integer variable with the name `years` can be declared like this:
+
+```cpp
+int years;
+```
+
+It is good practice to initialize variables upon declaration.
+C++ offers different mechanisms to do so.
+The version with the curly braces is more in line with modern C++, but the equal-sign version is also very common.
+
+```cpp
+int tomatoes{80};
+int potatoes = 40;
+```
+
+```exercism/caution
+C++ does allow using uninitialized variables.
+Until the variable is deliberately set, it is undefined and might contain anything.
```

Comment:
suggestion: Until the value is deliberately set you are not allowed to access it.

Reply:
you are allowed, but it is undefined to read it. assigning to it is very well defined.

Reply:
what's the difference between "this is not allowed" and "if you do that you get UB"? Or are we splitting hairs here?

Reply:
I'm hair-splitting on what `variable access` means. Memory write and memory read both are access in my dictionary but only the write is allowed. you are disallowed to read uninitialized memory to get a well formed program.

on the contrary
```cpp
void do_nothing(int&) {}
int* danger = nullptr;
int main() {
    do_nothing(*danger);
}```
does this access contents in the memory? syntactically we just dereferenced a null pointer.

Reply:
Where I live the law does not forbid squaring the circle. However, should I one day do it, then who knows what will happen.

But I don't think we should burden the student with the difference. What is important to know is that _it is possible to misbehave_ and also that _you shouldn't misbehave, lest bad things happen_.

I don't care for the «undefined behavior» terminology. In C++ land there are multiple flavors of it (though they go by different names), and outside of it sometimes different phenomena are meant, not always easily mapped onto C++ concepts.

Reply:
> What is important to know is that it is possible to misbehave and also that you shouldn't misbehave, lest bad things happen.

This is what we need to communicate!

I hate when teaching materials lie about some concepts because it is too complicated to explain correctly. If we simplify things it has to be explicitly stated that it is just a friendly little analogy that helps the understanding, but not the whole picture.

Reply:
That's why I wrote "value". An uninitialized *variable* has an indeterminate *value*. You are allowed to assign something to that variable but not to access the value. But I see why that can be confusing.

Reply:
I want to be more strict than "it is possible to misbehave". With UB you get worse things than random data, the whole program can do something completely unexpected. That's why I used the informal term "not allowed".

Reply:
I know better, but am still vulnerable to interpreting «is not allowed» as meaning _some compiler error will keep me from doing it_. Other people are likely to misintrepret it this way as well.

In https://github.com/exercism/cpp/pull/594#discussion_r1172714482 I tried to avoid the overly technical issues.